File: src/Samples/Moq.VisualBasic/Class1.vb
Namespace Sample.CSharp

    Public Class Class1

        Public Sub Test()
            Dim formatter As ICustomFormatter = Mock.Of(Of ICustomFormatter, IDisposable)
            Dim foo = Mock.Of(Of IFoo)
            Dim bar = Mock.Of(Of IBar)
            Dim value = "foo"

            foo.Id _
               .Callback(Sub() value = "before") _
               .Returns(Function() value) _
               .Callback(Sub() value = "after") _
               .Returns(Function() value)

            Console.WriteLine(foo.Id)
            Console.WriteLine(foo.Id)
        End Sub

    End Class
End Namespace

Public Interface IBar

    Sub DoBar()

End Interface

Public Interface IFoo

    Sub [Do]()

    Property Id As String

    Property Title As String

End Interface

File: src/Samples/Moq.CSharp/Class1.cs
using System;
using Moq;

namespace Sample.CSharp
{
    public class Class1
    {
        public void Test()
        {
            var mock = Mock.Of<CalculatorBase, ICustomFormatter, IDisposable>();

            mock.Setup(m => m.Memory.Recall()).Returns(1);

            var foo = Mock.Of<IFoo>();
            var bar = Mock.Of<IBar>();

            var value = "foo";

            foo.Id
                .Callback(() => value = "before")
                .Returns(() => value)
                .Callback(() => value = "after")
                .Returns(() => value);

            Console.WriteLine(foo.Id);
            Console.WriteLine(foo.Id);
        }

        public void Test1()
        {
            var fake = Mock.Of<ICalculator>();

            fake.Memory.Recall().Returns(5);
        }

        public void Test2()
        {
            var fake = Mock.Of<ICalculator>();

            fake.Memory.Recall().Returns(5);
        }
    }
}

public interface IBar
{
    void DoBar();
}

public interface IFoo
{
    void Do(bool donow);

    string Id { get; }
    string Title { get; set; }
}